src/com/chess/Game/ChessBoard.java

```diff
@@ -53,51 +53,87 @@ protected ChessBoard(Piece[][] board, NamedColor p1Color, NamedColor p2Color) {
         this.p2Color = p2Color;
     }
 
-    public List<Move> getAllMoves(NamedColor color) {
+    public List<Move> getAllMoves(NamedColor color, boolean forCheck) {

```

Comment:
This forCheck variable is in place to stop an infinite recursion. Since a player can't make a move that puts them in check it is impossible for both to be in check at the same time so if we are checking if someone is in check we assume the other is not using this flag.